File: flutteroutfity/flutteroutfity/lib/models/coupon.dart
class Coupon {
  final String id;
  final String code;
  final String type;
  final double value;
  final double minOrder;
  final bool isActive;
  final String? expiresAt;

  Coupon({
    required this.id,
    required this.code,
    required this.type,
    required this.value,
    this.minOrder = 0,
    this.isActive = true,
    this.expiresAt,
  });

  factory Coupon.fromJson(Map<String, dynamic> json) {
    return Coupon(
      id: json['_id'] ?? '',
      code: json['code'] ?? '',
      type: json['type'] ?? 'percentage',
      value: (json['value'] ?? 0).toDouble(),
      minOrder: (json['minOrder'] ?? 0).toDouble(),
      isActive: json['isActive'] ?? true,
      expiresAt: json['expiresAt'],
    );
  }
}

File: backend/models/Coupon.js
const mongoose = require('mongoose');

const couponSchema = new mongoose.Schema({
  code: {
    type: String,
    required: [true, 'Please enter coupon code'],
    unique: true,
    uppercase: true,
    trim: true,
  },
  type: {
    type: String,
    enum: ['percentage', 'fixed', 'free_shipping'],
    required: [true, 'Please select coupon type'],
  },
  value: {
    type: Number,
    required: [true, 'Please enter coupon value'],
    min: [0, 'Value cannot be negative'],
  },
  minOrder: {
    type: Number,
    default: 0,
  },
  maxUses: {
    type: Number,
    default: 100,
    min: 1,
  },
  usedCount: {
    type: Number,
    default: 0,
  },
  isActive: {
    type: Boolean,
    default: true,
  },
  expiresAt: {
    type: Date,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Coupon', couponSchema);
